src/scheduler/node_info.c

```diff
@@ -459,6 +469,12 @@ query_node_info(struct batch_status *node, server_info *sinfo)
 		}
 		attrp = attrp->next;
 	}
+	if (check_expiry) {
```

Comment:
I think we could move this if block to a place at line number 394 i.e. as soon as we get the value of the variable expiry.

Reply:
We can do this check only after the while loop as the ordering of attribute is not defined.